docs/build/wasm/transaction-fees.md

```diff
@@ -0,0 +1,147 @@
+It also incentives users to remove unused data from the chain by getting rent fees back. Any user can get back the rent fees if they remove on-chain data (not specifically the user that was first charged for).
+
+### Storage Rent Calculation
+
+This fee is calculated with the price set for each storage item `DepositPerItem`, and for each byte of storage `DepositPerByte`. In Astar, the deposit fee are defined as follows (more detail is this [Astar forum post](https://forum.astar.network/t/revising-rent-fee-on-astar-shiden/4309/1)):
+
+| Deposit Type | Shiden               | Astar              |
+|--------------|----------------------|--------------------|
+| Per Storage Byte | 0.00002 SDN | 0.002 ASTR |
+| Per Storage Item |  	0.004 SDN   | 0.4 ASTR |
+
+##### Calculation
+
+When a user stores a new key/value in a `Mapping` field, one `DepositPerItem` is charged. The length in bytes of the value is added also added to the fee (so bytes length x `DepositPerByte`).     
+For example, if a user store a new entry in a `Mapping<u32, AccountId>` (`AccountId` is 32 bytes) it will be charged `DepositPerItem` + 32 x `DepositPerByte`. 
+
+### What does it mean ?
+
+#### For users
+
+The first call to a dApp (one or several's smart contracts) will usually be more expensive than the following ones.
+This is because the first call will create a lot of new entries for the `AccountId`. From the second call it should be way cheaper (or free) because it will just update those items.
+
+As this is a `reserved balance`, user can't spend it. 
+If a user want to get it back, it should remove on-chain data. It is only possible if the smart-contract expose a function that remove data from chain (like `remove_mapping_entry` in the example below).
+
+#### For smart-contracts developers
+
+As the only way for users to get back their reserved balance is to remove on-chain data, it is important to make sure that the smart-contract expose functions that allow users to do so.   
+If the contracts don't expose such functions, there will be no way to remove on-chain data used by the contract and the 
+users will not be able to get back their reserved balance back (as it will be reserved balance on the contract account).
+
+### StorageDepositLimit
+
+When doing a contract call one of the argument is `StorageDepositLimit`. This value is the maximum amount of storage rent that can be charged for a single call.    
+If it's set to `None` it allow contracts to charge arbitrary amount of funds to be drained from the caller's account. 
+So it is necessary to set a limit (first dry-run the call to get the storage deposit amount) to prevent malicious contracts from draining funds from a user's account.
+This especially applies for front end applications that triggers contracts calls.
```

Comment:
To follow-up on the previous comment, something like:

"As user you are responsible for ensuring gas limit & storage deposit limits are sensible when interacting with smart contracts. This is same as for EVM smart contracts, but instead of having only non-refundable _gas_, you also have to take note of `storage_deposit_limit`."